pkg/utils/kubernetes/kubernetes.go

```diff
@@ -317,8 +311,36 @@ func ReconcileServicePorts(existingPorts []corev1.ServicePort, desiredPorts []co
 	return out
 }
 
-func buildEventsErrorMessage(events []corev1.Event) string {
-	sortByLastTimestamp := func(o1, o2 controllerutil.Object) bool {
+// FetchEventMessages gets events for the given object of the given `eventType` and returns them as a formatted output.
+// The function expects that the given `obj` is specified with a proper `metav1.TypeMeta`.
+func FetchEventMessages(ctx context.Context, c client.Client, obj client.Object, eventType string, eventsLimit int) (string, error) {
+	apiVersion, kind := obj.GetObjectKind().GroupVersionKind().ToAPIVersionAndKind()
+	if apiVersion == "" {
+		return "", fmt.Errorf("apiVersion not specified for object %s/%s", obj.GetNamespace(), obj.GetName())
+	}
+	if kind == "" {
+		return "", fmt.Errorf("kind not specified for object %s/%s", obj.GetNamespace(), obj.GetName())
+	}
+	fieldSelector := client.MatchingFields{
+		"involvedObject.apiVersion": apiVersion,
```

Comment:
@ialidzhikov this was added after talking to @rfranzke and @timebertt. Can you confirm that this was not left out by intention earlier?

Reply:
Yes, it makes sense to also include the apiVersion. 👍